src/core/core.datasetController.js

```diff
@@ -156,8 +156,22 @@ module.exports = function(Chart) {
 			model.backgroundColor = custom.hoverBackgroundColor ? custom.hoverBackgroundColor : valueOrDefault(dataset.hoverBackgroundColor, index, getHoverColor(model.backgroundColor));
 			model.borderColor = custom.hoverBorderColor ? custom.hoverBorderColor : valueOrDefault(dataset.hoverBorderColor, index, getHoverColor(model.borderColor));
 			model.borderWidth = custom.hoverBorderWidth ? custom.hoverBorderWidth : valueOrDefault(dataset.hoverBorderWidth, index, model.borderWidth);
-		}
-	});
+		},
+		//gets the max border or hover width to properly scale pie charts
+        getMaxBorderWidth: function (elements) {
```

Comment:
Since this is only for pie charts, would it make sense to add it in the doughnut controller instead of the base class?